powerline/bindings/bar/powerline-bar.py

```diff
@@ -35,18 +31,37 @@ def render(event=None, data=None, sub=None):
 		help='Subscribe for i3 events.'
 	)
 	args = parser.parse_args()
-	powerline = BarPowerline()
-
-	interval = 0.5
+	line = BarPowerline()
 	lock = Lock()
-
 	write = get_unicode_writer(encoding='utf-8')
 
+	def render(reschedule=False):
+		if reschedule:
+			Timer(0.5, render, kwargs={"reschedule": True}).start()
+
+		global lock
+		with lock:
+			write(line.render())
+			write('\n')
+			sys.stdout.flush()
+
+	def update(mode):
+		line.mode = mode
```

Comment:
I would highly suggest not to use Powerline class state. You have a number of closures here, so it is fine to save this argument in `d` dictionary:

```Patch
@@ -34,10 +34,11 @@
 	line = BarPowerline()
 	lock = Lock()
 	write = get_unicode_writer(encoding='utf-8')
+	mode_list = [None]
 
 	def render(reschedule=False):
 		if reschedule:
-			Timer(0.5, render, kwargs={"reschedule": True}).start()
+			Timer(0.5, render, kwargs={"reschedule": True, 'mode': mode_list[0]}).start()
 
 		global lock
 		with lock:
@@ -46,7 +47,7 @@
 			sys.stdout.flush()
 
 	def update(mode):
-		line.mode = mode
+		mode_list[0] = mode
 		render()
 
 	render(reschedule=True)
```